modulo3/projeto-astromatch/projeto-astromatch/src/components/MatchPage.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react";
+import axios from 'axios';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faFire } from "@fortawesome/free-solid-svg-icons";
+import { faHome } from "@fortawesome/free-solid-svg-icons";
+import { faPerson } from "@fortawesome/free-solid-svg-icons";
+import { createGlobalStyle } from 'styled-components';
+import Logo from '../img/logo.png.png'
+import Home from '../img/home.png.png'
+import BrokenHeart from '../img/broken-heart.png.png'
+import {
+    ImgHome, ImgLogo, DivPai, Button, DivContainer, DivCaixinha,
+    DivImg, ImgMatch, PersonMatch, ImgBrokenHeart
+} from './stylesMatch'
+
+const GlobalStyle = createGlobalStyle`
+* {
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0px;
+}
+
+// const ImgBrokenHeart = styled.img`
+// height: 100px;
+// width: 100px;
+// display: flex;
+// justify-content:center;
+// align-items: center;
+// margin-top: 200px;
+// margin-left: 150px;
+// margin-bottom: 20px;
+
+// `
+
+const url = "https://us-central1-missao-newton.cloudfunctions.net/astroMatch/Maria-Eduarda-Lopes-Silveira/matches"
+
+
+function MatchPage(props) {
+    const [matches, setMatches] = useState([])
+
+    useEffect(() => {
+        getMatches()
+    }, [matches])
```

Comment:
getMatches vai ser chamado todas as vezes que o setMatchs for chamado, e isso cria um loop, porque getMatches  é uma função que chama o setMatchs